src/ServiceClient.js

```diff
@@ -0,0 +1,184 @@
+import grpc, { InterceptingCall } from "grpc";
+import url from "url";
+import { BigNumber } from 'bignumber.js';
+import { find, map } from 'lodash';
+import paymentChannelStateServices from './payment_channel_state_service_grpc_pb';
+
+export default class ServiceClient {
+  constructor(sdk, metadata, group, ServiceStub, paymentChannelManagementStrategy, options = {}) {
+    this._sdk = sdk;
+    this._options = options;
+    this._metadata = metadata;
+    this._group = group;
+    this._paymentChannelManagementStrategy = paymentChannelManagementStrategy;
+    this._grpcStub = this._generateGrpcStub(ServiceStub);
+    this._paymentChannelStateServiceClient = this._generatePaymentChannelStateServiceClient();
+    this._paymentChannels = [];
+  }
+
+  get stub() {
+    return this._grpcStub;
+  }
+
+  get group() {
+    return this._group;
+  }
+
+  get groupIdInBytes() {
+    return Buffer.from(this.group.group_id, 'base64');
+  }
+
+  get paymentChannels() {
+    return this._paymentChannels;
+  }
+
+  get metadata() {
+    return this._metadata;
+  }
+
+  get paymentAddress() {
+    return this.group.payment_address;
+  }
+
+  get paymentChannelStateServiceClient() {
+    return this._paymentChannelStateServiceClient;
+  }
+
+  async loadOpenChannels() {
+    const currentBlockNumber = await this._web3.eth.getBlockNumber();
+    const newPaymentChannels = await this._mpeContract.getPastOpenChannels(this._account, this, this._lastReadBlock);
+    this._paymentChannels = [...this._paymentChannels, ...newPaymentChannels];
+    this._lastReadBlock = currentBlockNumber;
+    return this._paymentChannels;
+  }
+
+  async updateChannelStates() {
+    const currentChannelStatesPromise = map(this._paymentChannels, (paymentChannel) => {
+      return paymentChannel.syncState();
+    });
+    await Promise.all(currentChannelStatesPromise);
+    return this._paymentChannels;
+  }
+
+  async defaultChannelExpiration() {
```

Comment:
This method could be a private method of the `PaymentChannelManagementStrategy`, `expiryThreshold` which is required to calculate the expiration is available via public `metadata` method.

Reply:
Addressed in [commit](https://github.com/singnet/snet-sdk-js/pull/1/commits/aaa2e04f9158ded2f42774aaf9da6939875fc042)